src/Entities/Entity.cpp

```diff
@@ -1522,13 +1550,23 @@ bool cEntity::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
 	}
 
 	// Queue add to new world
-	a_World->AddEntity(this);
 	cWorld * OldWorld = cRoot::Get()->GetWorld(GetWorld()->GetName());  // Required for the hook HOOK_ENTITY_CHANGED_WORLD
 	SetWorld(a_World);
 
 	// Entity changed the world, call the hook
 	cRoot::Get()->GetPluginManager()->CallHookEntityChangedWorld(*this, *OldWorld);
 
+	OldWorld->QueueTask([this, ParentChunk, a_World](cWorld & a_OldWorld)
+	{
+		LOGD("Warping entity #%i (%s) from world (%s) to (%s). Source chunk: (%d, %d) ",
```

Comment:
I'd prefer world names in dblquotes `\"%s\"`; the server uses kinda convention of listing block coords in curly braces and chunk coords in square brackets.